android/src/main/java/com/clevertap/react/CleverTapEventEmitter.kt

```diff
@@ -0,0 +1,122 @@
+package com.clevertap.react
+
+import android.util.Log
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import java.util.LinkedList
+import java.util.Queue
+
+/**
+ * CleverTapEventEmitter is responsible for emitting events to the React Native JavaScript layer.
+ * It manages event buffers and allows events to be queued which is useful when needing to send an
+ * event while the ReactContext of the application is not yet initialized. Only events specified as
+ * [CleverTapEvent.bufferable] will be considered for buffering, all other events will be emitted
+ * immediately.
+ */
+object CleverTapEventEmitter {
+    private const val LOG_TAG = "CleverTapEventEmitter"
+
+    var reactContext: ReactContext? = null
+
+    private class Buffer(var enabled: Boolean) {
+        val items: Queue<Any?> by lazy { LinkedList() }
```

Comment:
lets add the thread mode here explicitly.

Reply:
Ok, I will add it.